File: scala/src/algorithms/warmup/DiagonalDifference.scala
package algorithms.warmup

object DiagonalDifference {
  def main(args: Array[String]): Unit = {
    val n = scala.io.StdIn.readLine().toInt
    val a = List.range(0, n).map(_ => scala.io.StdIn.readLine()).map(x => x.split(" ").map(_.toInt))
    val diag1 = List.range(0, n).map(x => a(x)(x)).sum
    val diag2 = List.range(0, n).map(x => a(x)((n - 1) - x)).sum
    println(Math.abs(diag1 - diag2))
  }
}

File: python/algorithms/warmup/diagonal_difference.py
n = int(input().strip())
a = []

for _ in range(n):
    temp = [int(row) for row in input().strip().split(' ')]
    a.append(temp)

diag_1 = sum([a[x][x] for x in range(n)])
diag_2 = sum([a[x][(n - 1) - x] for x in range(n)])

print(abs(diag_1 - diag_2))
